lib/analytics/course_students_csv_builder.rb

```diff
@@ -5,9 +5,14 @@
 class CourseStudentsCsvBuilder
   def initialize(course)
     @course = course
+    @created_articles = Hash.new(0)
+    @edited_articles = Hash.new(0)
+    @revisions = @course.all_revisions.where(new_article: true).pluck(:article_id, :user_id).to_h
```

Comment:
This looks like an efficient way to find the user who creates each new article. Name the instance variable more clearly, eg, `@new_article_revisions`.

Reply:
I'll get this done, should I also move this to generate_csv method? So all the computation happens there instead of some happening in new?

Reply:
Sure, seems reasonable. Either way would be fine with me.